plugins/pulp_rpm/plugins/importers/yum/sync.py

```diff
@@ -153,12 +151,14 @@ def run(self):
             return report
 
         except Exception, e:
-            _logger.exception('sync failed')
             for step, value in self.progress_status.iteritems():
                 if value.get('state') == constants.STATE_RUNNING:
                     value['state'] = constants.STATE_FAILED
                     value['error'] = str(e)
             self.set_progress()
+            # this is the new preferred way
+            if isinstance(e, PulpCodedException):
```

Comment:
This workflow does require lines 154-158 to run regardless of which kind of exception is caught. If I make a separate except block for ``PulpCodedException``, I'd need to factor out those lines, presumably to a separate function, and call it from both places. I'm not sure that's worth the trouble.